boba-chain-ops/node/node_mock.go

```diff
@@ -0,0 +1,84 @@
+package node
+
+import (
+	"errors"
+	"math/big"
+
+	"github.com/ledgerwatch/erigon-lib/common"
+	"github.com/ledgerwatch/erigon/cmd/rpcdaemon/commands"
+	"github.com/ledgerwatch/erigon/core/types"
+)
+
+var (
+	ErrGetLatestBlock = errors.New("GetLatestBlock mock rpc error")
+	ErrGetBlockByNum  = errors.New("GetBlockByNumber mock rpc error")
+	ErrGetTxByHash    = errors.New("GetTransactionByHash mock rpc error")
+	ErrForkchoice     = errors.New("ForkchoiceUpdateV1 mock rpc error")
+	ErrGetPayload     = errors.New("GetPayloadV1 mock rpc error")
+	ErrNewPayload     = errors.New("NewPayloadV1 mock rpc error")
+)
+
+type MockRPC struct {
```

Comment:
I'm always a little wary of bespoke mocks.  Did you consider using something like [counterfeiter](https://github.com/maxbrunsfeld/counterfeiter) to generate this?  It's another dep, but, would be less code to maintain.

No need to change, if you'd rather not, but just wanted to throw it out there.